manifests/config.pp

```diff
@@ -1,7 +1,19 @@
 # == Class chocolatey::config
-#
-# This class is called from chocolatey for service config.
-#
-# NOTE: CURRENTLY NON-FUNCTIONAL
 class chocolatey::config {
+  assert_private()
+
+  #todo: check choco version from custom choco_version fact
+
+  $_choco_exe_path = "${chocolatey::choco_install_location}\\bin\\choco.exe"
+
+  $_enable_autouninstaller = $chocolatey::enable_autouninstaller ? {
+    false => 'disable',
+    default => 'enable'
+  }
+
+  exec { "chocolatey_autouninstaller_${_enable_autouninstaller}":
+    path    => $::path,
+    command => "${_choco_exe_path} feature -r ${_enable_autouninstaller} -n autoUninstaller",
+    unless  => "${_choco_exe_path} feature list -r | findstr /X /I \"autoUninstaller.*\\[${_enable_autouninstaller}d\\]\"",
```

Comment:
Yes, that is exactly what Puppet is doing.... here's the chocolatey.log:

~~~
2015-08-27 01:42:43,458 [DEBUG] - Command line: "C:\ProgramData\chocolatey\choco.exe" feature list -r | findstr /X /I "autoUninstaller.*\[enabled\]"
2015-08-27 01:42:43,458 [DEBUG] - Received arguments: feature list -r | findstr /X /I autoUninstaller.*\[enabled\]
2015-08-27 01:42:43,489 [ERROR] - A single features command must be listed. Please see the help menu for those commands
2015-08-27 01:42:43,489 [DEBUG] - Exiting with 1
~~~